### PR TITLE
Fixed the phone number input box in "My Profile"

### PR DESCRIPTION
**Pull request by Boorclark**
Fixes Issue #615 

Changes Made:
 - Updated UI to match the SLC proposal
 - Fixed the backend for the Edit/Save

### CODE REVIEW

File: app/static/js/volunteerProfile.js [hunk 1]
```diff
@@ -125,27 +125,40 @@ $(document).ready(function(){
...
+        var phoneInput = $("#phoneInput");
+        let isvalid = phoneInput.val().replace(/\D/g,"").length === 10;
+        let isempty = phoneInput.val().replace(/\D/g,"").length === 0;
+        if (!(isvalid || isempty)) {
+            phoneInput.addClass("invalid");
+            window.setTimeout(() => phoneInput.removeClass("invalid"), 1000);
+            phoneInput.focus()
+            return false;
+        }
+        $(this).html('Edit');
```
Comment (BrianRamsay) [inline app/static/js/volunteerProfile.js]: Phone number handling is now duplicated. Let's make a function that these two places can share.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Make sure you are testing your code thoroughly. The `invalid` class isn't doing anything on this page.

File: app/static/js/base.js [hunk 2]
```diff
@@ -16,3 +16,44 @@ $(document).ready(function() {
...
+            },
+            error: function(request, status, error) {
+              msgFlash("Error updating phone number", "danger")
+            }
+          })
+      } else if (isvalid == false){
+          msgFlash("Invalid Phone number", "danger")
+        }
+
+        $('#phoneInput').inputmask('(999)-999-9999');
```
Comment (BrianRamsay) [inline app/static/js/base.js]: You don't need to do the inputmask step more than once when the page loads

File: app/static/js/base.js [hunk 3]
```diff
@@ -16,3 +16,44 @@ $(document).ready(function() {
...
+    var isvalid = null
+    if ($(editButtonId).html() === 'Edit') {
+        $(editButtonId).html("Save");
+        $(phoneInputId).focus();
+    } else {
+        // Save the phone number
+        var phoneInput = $(phoneInputId);
+        isvalid = phoneInput.val().replace(/\D/g,"").length === 10;
+        let isempty = phoneInput.val().replace(/\D/g,"").length === 0;
+        if (!(isvalid || isempty)) {
```
Comment (BrianRamsay) [inline app/static/js/base.js]: Maybe we should have a comment here clarifying that this expression allows the phone input to be empty.

File: app/static/js/base.js [hunk 4]
```diff
@@ -16,3 +16,44 @@ $(document).ready(function() {
...
+                  "phoneNumber":phoneInput.val()},
+            success: function(s){
+              msgFlash("Phone Number is updated", "success")
+            },
+            error: function(request, status, error) {
+              msgFlash("Error updating phone number", "danger")
+            }
+          })
+      } else if (isvalid == false){
+          msgFlash("Invalid Phone number", "danger")
```
Comment (BrianRamsay) [inline app/static/js/base.js]: This could just go up with the other things that happen when it is invalid

File: app/static/js/base.js [hunk 5]
```diff
@@ -16,3 +16,44 @@ $(document).ready(function() {
...
+        isvalid = phoneInput.val().replace(/\D/g,"").length === 10;
+        let isempty = phoneInput.val().replace(/\D/g,"").length === 0;
+        if (!(isvalid || isempty)) {
+            phoneInput.addClass("invalid");
+            window.setTimeout(() => phoneInput.removeClass("invalid"), 1000);
+            phoneInput.focus()
+            isvalid = false;
+            return isvalid;
+        }
+        if (isvalid == true){
```
Comment (BrianRamsay) [inline app/static/js/base.js]: We want to be able to save an empty phone number, so we can just get rid of this isvalid check.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

File: app/templates/main/volunteerProfile.html [hunk 6]
```diff
@@ -17,8 +17,8 @@
     <h1 align="center">{{volunteer.firstName}} {{volunteer.lastName}}</h1>
     <h6 align="center">{{volunteer.email}}</h6>
     <div align="center">
-          <input size="14" id="phoneInput" type="text"  class="bfh-phone" value="{{volunteer.phoneNumber}}">
-          <button class=" btn-success btn-sm" id="updatePhone" data-username="{{volunteer.username}}">Update</button>
+          <input type="text" style="border: none" size="12" class="form-control-sm bfh-phone"  id="phoneInput" value="{{volunteer.phoneNumber}}"/>
```
Comment (zachneill) [inline app/templates/main/volunteerProfile.html]: There's a bootstrap equivalent for "border:none"

File: app/static/js/slcNewProposal.js [hunk 7]
```diff
@@ -84,37 +84,9 @@ $(document).ready(function(e) {
...
-                  contentType: "application/json",
-                  success: function(response) {
-                      msgFlash("Instructor's phone number updated", "success")
-                  },
-                  error: function(request, status, error) {
-                    msgFlash("Error updating phone number", "danger")
-                  }
-                });
-            }
+            var phoneInput = "#inputPhoneNumber-" + username
```
Comment (zachneill) [inline app/static/js/slcNewProposal.js:87]: It's 2022 stop using var 

Either use const or let, or clock out and go back to gardening and watching The Price is Right.

File: app/static/js/base.js [hunk 8]
```diff
@@ -16,3 +16,37 @@ $(document).ready(function() {
...
+        $(phoneInputId).focus();
+    } else {
+        // Save the phone number
+        var phoneInput = $(phoneInputId);
+        var isvalid = phoneInput.val().replace(/\D/g,"").length === 10;
+        let isempty = phoneInput.val().replace(/\D/g,"").length === 0;
+        if (!(isvalid || isempty)) { // allows phone number input to be empty
+            phoneInput.addClass("invalid");
+            window.setTimeout(() => phoneInput.removeClass("invalid"), 1000);
+            phoneInput.focus()
```
Comment (zachneill) [inline app/static/js/base.js:32]: Missing semicolons in this file. Also stop using var, grandma.

**Review by BrianRamsay — APPROVED** [review]
Nice!

File: app/static/js/base.js (hunk 8 above)
Comment (zachneill) [inline app/static/js/base.js:32]: Missing semicolons in this file. Also stop using var, grandma.

**Comment by zachneill** [comment]
Commented after this was merged smh

**Pull request merged**